packages/react-components/react-table/src/hooks/useKeyboardResizing.ts

```diff
@@ -54,57 +54,85 @@ export function useKeyboardResizing(columnResizeState: ColumnResizeState) {
       case Enter:
       case Escape:
         stopEvent();
-        disableInteractiveMode();
+        // Just blur here, the onBlur handler will take care of the rest (disableInteractiveMode).
+        refMap.get(columnId)?.current?.blur();
         break;
     }
   });
 
-  // On component unmout, cancel any timer for adding a listener (if it exists) and remove the listener
-  React.useEffect(
-    () => () => {
-      targetDocument?.defaultView?.clearTimeout(addListenerTimeout.current);
-      targetDocument?.defaultView?.removeEventListener('keydown', keyboardHandler);
-    },
-    [keyboardHandler, targetDocument?.defaultView],
-  );
-
   const enableInteractiveMode = React.useCallback(
     (colId: TableColumnId) => {
       setColumnId(colId);
       onChangeRef.current?.(colId, true);
-      // Create the listener in the next tick, because the event that triggered this is still propagating
-      // when Enter was pressed and would be caught in the keyboardHandler, disabling the keyboard mode immediately.
-      // No idea why this is happening, but this is a working workaround.
-      // Tracked here: https://github.com/microsoft/fluentui/issues/27177
-      addListenerTimeout.current = targetDocument?.defaultView?.setTimeout(() => {
-        targetDocument?.defaultView?.addEventListener('keydown', keyboardHandler);
-      }, 0);
+      // This needs to happen asynchronously, because we cannot focus an element with without tabIndex.
+      // Only after we have called "setColumnId" the tabIndex will be set to 0 for this handle.
+      // Meaning we have to wait until the DOM updates with the new tabIndex and then we can focus.
+      setTimeout(() => {
```

Comment:
Similarly here - I'm not sure if a timeout here is a good idea. There is already imperative code, so there should be no problem setting the tabindex here using the ref

Reply:
The `tabindex` is provided in `getKeyboardResizingProps`, by default it is undefined because we don't want the resize handle to be a part of tab (or arrow) navigation by default. Enabling the resize mode sets `currentColumnId` to the right column id, which results in the resize handle receiving the tabIndex=0, for it to be actually able to take focus. Then we can focus it after the render.

How would I set the tabIndex using the ref to keep it consistent across re-renders?

Reply:
I refactored to this:

```ts
const enableInteractiveMode = React.useCallback(
    (colId: TableColumnId) => {
      setColumnId(colId);
      onChangeRef.current?.(colId, true);
      const handle = refMap.get(colId)?.current;
      if (handle) {
        handle.tabIndex = 0;
        handle.focus();
      }
    },
    [refMap],
  );
```

and it still seems to work

Reply:
I was a bit worried that re-rendering the header would result in losing the focus, but that doesn't seem to be the case. I'll go this route :)